### PR TITLE
pocket_lab: Added Extensions

### PR DESCRIPTION
**Pull request by D-Disha**
Added extension callbacks needed for application layer. Added structure member to extend views.

### CODE REVIEW

File: pocket_lab/pl_gui_views.c [hunk 1]
```diff
@@ -177,6 +193,15 @@ static adi_fft_data_to_volt_conv data_to_volt_wrt_vref;
  * binary data */
 static adi_fft_code_to_straight_bin_conv code_to_straight_binary;
 
+/* Pocket Lab Instance */
+static struct pl_gui_init_param pl_instance =  {
+	.event = NULL,
+	.event1 = NULL
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:199]: what is this event1?

File: pocket_lab/pl_gui_views.c [hunk 2]
```diff
@@ -177,6 +193,15 @@ static adi_fft_data_to_volt_conv data_to_volt_wrt_vref;
...
+};
+
+/* Local copy of init params */
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:202]: should this be "global" copy?

File: pocket_lab/pl_gui_views.c [hunk 3]
```diff
@@ -881,32 +916,31 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
...
 								   LV_STATE_CHECKED | LV_STATE_DISABLED);
-					} else {
+					else
 						lv_obj_clear_state(pl_gui_dmm_chn_checkbox[cnt], LV_STATE_DISABLED);
-					}
 				}
 
 				lv_obj_set_style_bg_color(btn, lv_palette_main(LV_PALETTE_GREEN), LV_PART_MAIN);
 				pl_gui_dmm_is_running = !pl_gui_dmm_is_running;
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:934]: extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 4]
```diff
@@ -961,8 +996,10 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
 						lv_chart_remove_series(pl_gui_capture_chart_ovrly, pl_gui_capture_chn_ser[cnt]);
 						pl_gui_capture_chn_ser[cnt] = NULL;
 					} else {
-						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt],LV_STATE_DISABLED);
+						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt], LV_STATE_DISABLED);
+
 					}
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:1002]: extra line can be removed

**Review by jansunil — CHANGES_REQUESTED** [review]

File: pocket_lab/pl_gui_views.c (hunk 1 above)
Comment (D-Disha) [reply pocket_lab/pl_gui_views.c:199]: These are the events we need for the application layer. The events are when stop is pressed / when 2048 samples are complete. We don't want the reason why these events were added and exposed in the library as it was for a specific cbm use case. Hence, I have kept a very generic name.

File: pocket_lab/pl_gui_views.c (hunk 2 above)
Comment (D-Disha) [reply pocket_lab/pl_gui_views.c:202]: Local here means local copy needed for the file ie local copy of the init params passed from the application layer. HHave used the same comments for other such copies as well.  Do you suggest a better comment can be added?

File: pocket_lab/pl_gui_views.c (hunk 1 above)
Comment (SaikiranGudla) [reply pocket_lab/pl_gui_views.c:199]: Do you think it makes more sense with event1, event2 as event names?

File: pocket_lab/pl_gui_views.c [hunk 5]
```diff
@@ -881,15 +917,16 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
...
 			}
-		} else { // Start/Stop label
+		} else {
+			// Start/Stop label
 			if (pl_gui_dmm_is_running) {
 				lv_label_set_text(label, "Start");
 
 				for (cnt = 0; cnt < pl_gui_dmm_chn_cnt; cnt++) {
-					if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+					if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED)
```
Comment (SaikiranGudla) [inline pocket_lab/pl_gui_views.c:926]: nit: Missing braces

**Review by jansunil — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]

**Pull request merged**